src/IO/H5/VolumeData.cpp

```diff
@@ -18,10 +20,61 @@
 #include "IO/H5/Header.hpp"
 #include "IO/H5/Helpers.hpp"
 #include "IO/H5/Version.hpp"
-
+#include "Utilities/Numeric.hpp"
 /// \cond HIDDEN_SYMBOLS
 namespace h5 {
-VolumeData::VolumeData(const bool exists, detail::OpenGroup&& group,
+namespace {
+// Append the element extents and connectevity to the total extents and
+// connectivity
+void append_element_extents_and_connectivity(
+    gsl::not_null<std::vector<size_t>*> total_extents,
+    gsl::not_null<std::vector<int>*> total_connectivity,
+    gsl::not_null<int*> total_points_so_far, size_t dim,
+    ExtentsAndTensorVolumeData element) {
+  // Process the element extents
+  const auto& extents = element.extents;
+  if (extents.size() != dim) {
+    ERROR("Trying to write data of dimensionality"
+          << extents.size() << "but the VolumeData file has dimensionality"
+          << dim << ".");
+  }
+  total_extents->insert(total_extents->end(), extents.begin(), extents.end());
+  // Find the number of points in the local connectivity
+  const int element_num_points =
+      alg::accumulate(extents, 1, std::multiplies<>{});
+  // Generate the connectivity data for the element
+  const std::vector<int> connectivity =
+      [&extents, &total_points_so_far ]() noexcept {
+    std::vector<int> local_connectivity;
+    for (const auto& cell : vis::detail::compute_cells(extents)) {
+      for (const auto& bounding_indices : cell.bounding_indices) {
+        local_connectivity.emplace_back(*total_points_so_far +
+                                        static_cast<int>(bounding_indices));
+      }
+    }
+    return local_connectivity;
+  }
+  ();
+  *total_points_so_far += element_num_points;
+  total_connectivity->insert(total_connectivity->end(), connectivity.begin(),
+                             connectivity.end());
+}
+// Append the name of an element to the string of grid names
+void append_element_name(gsl::not_null<std::string*> grid_names,
+                         ExtentsAndTensorVolumeData element) {
+  // Get the name of the element
+  const auto& first_tensor_name = element.tensor_components.front().name;
+  ASSERT(first_tensor_name.find_last_of('/') != std::string::npos,
+         "The expected format of the tensor component names is "
+         "'GROUP_NAME/COMPONENT_NAME' but could not find a '/' in '"
+             << first_tensor_name << "'.");
+  const auto spatial_name =
+      first_tensor_name.substr(0, first_tensor_name.find_last_of('/'));
+  *grid_names += spatial_name + ":";
```

Comment:
Can we make the `separator` (`:`) a member function of the `VolumeData` class and pass it into all the places that need it? This will make changing what the separator is in the future much easier.

```cpp
class VolumeData {
  ....
 private:
  static char separator() noexcept { return ':'; }
 ...
};
```